airbyte-cdk/python/airbyte_cdk/sources/streams/http/http.py

```diff
@@ -34,11 +34,10 @@ class HttpStream(Stream, ABC):
     def __init__(self, authenticator: Union[AuthBase, HttpAuthenticator] = None):
         self._session = requests.Session()
 
-        self._authenticator = NoAuth()
+        self._authenticator = authenticator or NoAuth()
```

Comment:
main goal is this update is to make authenticator property be able to return all types of authenticator  objects (including AuthBase.

This authenticator object is needed to be used by substreams initiated from some main streams

Reply:
I will  bump cdk version and do additional tasks once this approach is approved

Reply:
@midavadim is there a real scenario for this? normally streams should not create other streams, see parent/child abstraction